functions/main.py

```diff
@@ -0,0 +1,130 @@
+# Welcome to Cloud Functions for Firebase for Python!
+# To get started, simply uncomment the below code or create your own.
+# Deploy with `firebase deploy`
+
+import logging
+from datetime import datetime
+from functools import reduce
+from io import StringIO
+from typing import Any, Dict, List
+
+import networkx as nx
+from bibx import Sap, read_any
+from firebase_admin import firestore, initialize_app, storage
+from firebase_functions.firestore_fn import (DocumentSnapshot, Event,
+                                             on_document_created)
+from firebase_functions.options import MemoryOption
+
+logging.basicConfig(level=logging.INFO)
+
+initialize_app()
+
+
+def tree_from_strings(strings: List[str]) -> nx.DiGraph:
+    """Creates a ToS tree from a list of strings."""
+    collections = [read_any(StringIO(text)) for text in strings]
+    collection = reduce(lambda x, y: x.merge(y), collections)
+    sap = Sap()
+    graph = sap.create_graph(collection)
+    graph = sap.clean_graph(graph)
+    return sap.tree(graph)
+
+
+def convert_tos_to_json(tree: nx.DiGraph) -> Dict[str, List[Dict]]:
+    """
+    Converts a ToS graph in the default format to be processed by the frontend.
+    """
+    output = {}
+    sections = ["root", "trunk", "leaf"]
+    for section in sections:
+        data = sorted(
+            [
+                {
+                    key: val
+                    for key, val in data.items()
+                    if not key.startswith("_") and key != "extra"
+                }
+                for node, data in tree.nodes.items()
+                if tree.nodes[node][section] > 0
+            ],
+            key=lambda article: article.get(section, 0),
+            reverse=True,
+        )
+        output[section] = data
+
+    return output
+
+
+def get_contents(
+    document_data: Dict[str, Any], max_size_megabytes=10
```

Comment:
Let's load the max size from an env var, which will be used here also `src/components/upload/errors.ts`, where the value of the max file size is defined in the front-end.